repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/PurgingEntityViewListener.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2018-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.repository.kafka;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import com.expediagroup.streamplatform.streamregistry.state.EntityView;
+import com.expediagroup.streamplatform.streamregistry.state.EntityViewListener;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.SpecificationDeletionEvent;
+import com.expediagroup.streamplatform.streamregistry.state.model.specification.Specification;
+
+@AllArgsConstructor
+class PurgingEntityViewListener implements EntityViewListener {
+  private final EntityView entityView;
+
+  @Override
+  public <K extends Entity.Key<S>, S extends Specification> void onEvent(Entity<K, S> oldEntity, @NonNull Event<K, S> event) {
+    if (event instanceof SpecificationDeletionEvent) {
+      entityView.purgeDeleted(event.getKey());
```

Comment:
Is this going to cause everything to be purged before an offline agent has a chance to see it?

Reply:
Yeah, thats the idea 😄 

This is currently only used by the `stream-registry` which has no use for knowing about deletes. See The conversation with @nahguam above (https://github.com/ExpediaGroup/stream-registry/pull/270#discussion_r561713684)

Reply:
Thanks for the offline discussion @jamespfaulkner. For the benefit of others, each agent has it's own copy of the entity view so this purge will not affect what they see.

Reply:
Remember this is only in-memory local to an app, this doesn't affect other apps. They will have their own in-memory view of the logically deleted (tombstoned) entities.